src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -120,7 +119,25 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
                     hasJsonBody = true;
                 }
 
-                apiDescription.ParameterDescriptions.Add(parameterDescription);
+                apiDescription.ParameterDescriptions.Add(parameterDescription!);
+            }
+
+            // Get custom attributes for the handler. ConsumesAttribute is one of the examples. 
+            var methodAttributes = methodInfo.GetCustomAttributes();
+            foreach (var attribute in methodAttributes)
+            {
+                ApiParameterDescription? parameterDescription;
+
+                if (attribute is IAcceptsMetadata)
+                {
+                    parameterDescription = CreateConsumesAttributeParameterDescription(attribute);
+                    if (parameterDescription is null)
+                    {
+                        continue;
+                    }
+
+                    apiDescription.ParameterDescriptions.Add(parameterDescription!);
+                }               
```

Comment:
This should use `routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>()`. We shouldn't be reflecting on the method since it does not include any metadata added via `.Accepts(...)`

```suggestion
            var acceptRequestType =  routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>()?.RequestType;
            if (acceptRequestType is not null)
            {
                var parameterDescriptor = new ApiParameterDescription
                {
                    Name = requestType.Name,
                    ModelMetadata = CreateModelMetadata(requestType),
                    Source = BindingSource.Body,
                    Type = requestType,
                    IsRequired = true,
                };
                
                apiDescription.ParameterDescriptions.Add(parameterDescriptor);
            }
```

Questions:

* What happens if the method already has a from-body parameter?
* How do we infer if the body is required?

Reply:
1. Swashbuckle only picks the first body parameter or default.  So, I suspect we should be fine if there are dupes. Shall we dedupe ?
2.  No easy way to infer optionality at the moment for Consumes/Accepts. How do we even create the NullabilityContext ?